### DISCUSSION TITLE
Feedback on running a tech meetup on hubs

### DISCUSSION THREAD

**Discussion by djay** [mozilla/hubs#3718] [General Questions & Discussion] [answered]
Yesterday we ran tech meetup in hubs with mixed success. I wanted to share the list of things that could be improved.
I'll create tickets for some of them.
I love the ideas of hubs, that its browser first, no downloads, works on all platforms. And is close to working well for a meetup.

Generally I think it worked better in some ways than zoom or discord and worse in others.
The main goal was to improve networking and discussion in the the breaks. I think that worked somewhat better in that multiple discussions happened at the same time naturally without the awkwardness of people jumping in and out of voice rooms in discord.

Thanks again for all the effort thats gone into this

# Presentations

- Presentations are hard to see. too small.
   - you can click the zoom/inspect button on the object menu but its not very discoverable. - https://github.com/mozilla/hubs/issues/3720
      - maybe should in the corner of object when click not under the space bar menu?
          - or at least for screen-share/webcam it should be 
          - or on hover - https://github.com/mozilla/hubs/issues/3128
      - maybe it should be a more standard full screen icon?
      - its not clear that it doesn't move you.
      - you can't tell when people are using this mode and not looking at you. 
          - Perhaps it should give the avatar binoculars or something?
      - I had no idea you could use right click to zoom/inspect 
   - you can use the menu top left with all the objects and click on the object. But it doesn't make the screen large enough. 
      -  maybe have the object menu on this view also so then you make it actually full screen?
      - why is this way of zooming different from the one on the object menu?
      - can’t see text messages when viewing this way
      - there is a goto button here which is yet another way to zoom in? but that takes you there?
- screen-share disappears for some
- perhaps would be nice if the speaker can combine webcam and screenshare at the same time
- speaker voice is off to one side. you have to get them to stand in the right place. not sure everyone heard them well. needs a microphone https://github.com/mozilla/hubs/issues/3404
- seats in the outdoor meetup confuse people since they don’t know why they can’t move. 
    - maybe jump keyboard shortcut might help? - https://github.com/mozilla/hubs/issues/3721
- speaker doesn’t get much reaction from people. can’t even tell if they are there - https://github.com/mozilla/hubs/issues/3689
   - show when someone is not looking at the screen? 
- some audio dropout. had to switch browsers or just grin and bear it
   - unclear if it was network or browser cpu issues    
- permissions 
   - with discord integration could not give screen-sharing or spawn permission just to speakers - https://github.com/mozilla/hubs/issues/3710
      - might be bug or perhaps I just couldn't work out the right discord permissions?
      - perhaps might be better if the admin creates the screenshare for the speaker and gives it to them? - https://github.com/mozilla/hubs/issues/3690
   - only work around was to turn it on for everyone. at least temporarily
   - if you leave the pen on anyone can use it to draw stuff. admin can't delete this

# Networking/Discussions

- can hear too many conversations or too distinctly. it was hard not to be distracted by some conversations happening nearby or in some cases further away. could fix with some of these?
  - more dropoff if more people are talking. https://github.com/mozilla/hubs/issues/2322
     - or rather than make it so the room is quieter turn it into a background chatter thats less distinct? a non directional chatter sound?  
  - people should speak in the direction they are facing.  https://github.com/mozilla/hubs/issues/3914
  - if people are between you and a speaker then it should block some of the sound. https://github.com/mozilla/hubs/issues/3914
  - if someone is behind you it should sound quieter. does it do this now? https://github.com/mozilla/hubs/issues/3914
  - https://github.com/mozilla/hubs/issues/1940 - increase dropoff for busy rooms
  - normalising speaker volumes. some peoples mics were louder than others - https://github.com/mozilla/hubs/issues/3214

- zombies. some people are away from the keyboard but you can't tell until you try to talk to them. could fix with
   - With face tracking you could show when someone are not there. like they become sleeping on the floor or face down or something. or get pushed out to lobby after a timeout.  - https://github.com/mozilla/hubs/issues/3689
   - could also track if they are looking at the room or some other app
- lots of mic problems. some had to restart the browser. some had to pic another mic in the lobby. some never spoke so we aren't sure if they were shy or had mic problems - https://github.com/mozilla/hubs/issues/2643
- sometimes got stream of leaves and joins from everyone in the room maybe due to socket disconnection?
- can’t reduce the music for everyone. not clear that turning it down didn’t turn it down for everyone. since the play button plays it for everyone shouldn’t volume be global too? - https://github.com/mozilla/hubs/issues/3397
   - You can only really move it away. 
- transition to a more social pub type venue for after the event doesn’t work well. hard to get people to follow you. if they don’t see where you went or heard you then they don’t know where to go. doesn’t feel seemless
   - maybe userlist should be grouped into rooms (when rooms that are linked togeather) so you can see at a glance where poeple are and follow them
   - text message says user leaving for X room with link
   - should be quicker loading and not answer same questions https://github.com/mozilla/hubs/issues/2606 
- avatars 
   - it's weird talking to snowmen and alligators for some kinds of events - 
   - it should be possible to make people stick to realistic avatars (like made by readyplayer.me)? more like spatial.io - https://github.com/mozilla/hubs/issues/3746

# Startup

- slow to load objects for some on entry
- people complained about CPU use. seems its less taxing on mobiles than laptops
- get "internal server error" when using discord authentication from older IOS and safari.
- cardboard didn't work for my phone at least. eye spacing was wrong and no way to pick which profile like some other cardboard sites - https://github.com/mozilla/hubs/issues/2558
- when using discordbot names are discord ids. It would be nice to be able to rename yourself since discord names are often not very useful when its a real meetup
- many were using thaikeyboards and wasd navigation doesn't seem to work when using a thai keyboard mode. https://github.com/mozilla/hubs/issues/2397

**Comment by robinkwilson** [comment, marked as answer]
This feedback is awesome thank you so much for this write up and the linked issues (and linking this discussion thread)! We definitely hear you and you're tracked on hubs issues!

I don't really have an answer, if you want the "discussion thread answered" credit, make a comment and either mark your comment as "Answered" or I can mark your comment as the answer.

**Reply by djay** [reply]
yeah just wanted to share for others who want to try something similar but github issues didn't see the right place.
I guess the question I was really asking myself (and still asking) is is it worth using it for a meetup vs discord/zoom/jitsi etc.
So far I've very divided. Presentations were worse and controls and mic problems got in the way of the content. But I felt networking portions went slightly better due to people more naturally having different discussions and migrating between them.

**Comment by djay** [comment]
Ran another one. This time issues were - still some people not being able to speak or hear.
and also people finding it hard to know who was speaking. (https://github.com/mozilla/hubs/issues/2533 could fix this but also perhaps just bigger mouths on certain avatars or facial tracking to show more natural expressions shown by someone speaker vs listening (https://github.com/mozilla/hubs/issues/3689)
and the new UI making fullscreen/inspect mode not full screen anymore. hard to see presentations https://github.com/mozilla/hubs/issues/3893
